modules/kubeconfig/main.tf

```diff
@@ -19,11 +19,25 @@ data "template_file" "kubeconfig_provider" {
   }
 }
 
+# 1.24 change: Create service account secret
+resource "kubernetes_secret" "sa_secret" {
+  count = var.create_static_kubeconfig ? 1 : 0
+  metadata {
+    name      = local.service_account_secret_name
+    namespace = var.namespace
+    annotations = {
+      "kubernetes.io/service-account.name" = local.service_account_name
+    }
+  }
+
+  type = "kubernetes.io/service-account-token"
+}
+
 # Service Account based kube config data/template/resources
 data "kubernetes_secret" "sa_secret" {
   count = var.create_static_kubeconfig ? 1 : 0
   metadata {
-    name      = kubernetes_service_account.kubernetes_sa.0.default_secret_name
+    name = "${kubernetes_secret.sa_secret[0].metadata.0.name}"
```

Comment:
Why variable syntax here?

Reply:
data source is removed in the latest change. Code now only refers the new resource for the secret.